presto-spi/src/main/java/com/facebook/presto/spi/block/DictionaryBlock.java

```diff
@@ -540,4 +542,23 @@ public boolean isNullUnchecked(int internalPosition)
         assert internalPositionInRange(internalPosition, idsOffset, positionCount);
         return dictionary.isNull(ids[internalPosition]);
     }
+
+    @Override
+    public Block appendNull()
+    {
+        int desiredLength = idsOffset + positionCount + 1;
+        int[] newIds = Arrays.copyOf(ids, desiredLength);
```

Comment:
elsewhere you are using ensureCapacity helper method; why not here?

Reply:
@mbasmanova This is because I didn't want to modify the original block. For DictionaryBlock, we always need to re-allocate ids array since we need to add one more element to it. If we modify the original ids array, then other block views that rely on this block get modified, which may cause problems. `ensureCapacity()` does not allocate new array if the array size is big enough so I just used `Arrays.copyOf()`.